src/elements/content-sharing/ContentSharing.js

```diff
@@ -0,0 +1,165 @@
+ * @author Box
+ */
+import React, { useCallback, useEffect, useState } from 'react';
+import { FormattedMessage } from 'react-intl';
+import type { $AxiosError } from 'axios';
+import API from '../../api';
+import Internationalize from '../common/Internationalize';
+import ErrorMask from '../../components/error-mask/ErrorMask';
+import UnifiedShareModal from '../../features/unified-share-modal';
+import messages from '../../features/unified-share-modal/messages';
+import { convertItemResponse, convertUserResponse } from '../../features/unified-share-modal/utils/convertData';
+import {
+    CLIENT_NAME_CONTENT_SHARING,
+    FIELD_ALLOWED_INVITEE_ROLES,
+    FIELD_ENTERPRISE,
+    FIELD_HOSTNAME,
+    FIELD_ID,
+    FIELD_NAME,
+    FIELD_TYPE as FIELD_ITEM_TYPE,
+    FIELD_SHARED_LINK,
+    FIELD_SHARED_LINK_FEATURES,
+    TYPE_FILE,
+    TYPE_FOLDER,
+    FIELD_PERMISSIONS,
+    FIELD_EXTENSION,
+    FIELD_DESCRIPTION,
+} from '../../constants';
+import { contentSharingMessages, CONTENT_SHARING_ERRORS } from './messages';
+import type { ErrorResponseData } from '../../common/types/api';
+import type { ItemType } from '../../common/types/core';
+import type { item as itemFlowType } from '../../features/unified-share-modal/flowTypes';
+import type { ContentSharingItemAPIResponse, ContentSharingSharedLinkType } from './types';
+
+type ContentSharingProps = {
+    apiHost: string,
+    itemID: string,
+    itemType: ItemType,
+    language: string,
+    showFormOnly?: boolean,
+    token: string,
+};
+
+function ContentSharing(props: ContentSharingProps) {
+    const { apiHost, itemID, itemType, language, token } = props;
+    const [item, setItem] = useState<?itemFlowType>(null);
+    const [sharedLink, setSharedLink] = useState<?ContentSharingSharedLinkType>(null);
+    const [currentUserID, setCurrentUserID] = useState<?string>(null);
+    const [errorMessage, setErrorMessage] = useState<?Object>(null);
+
+    const api = new API({
+        apiHost,
+        clientName: CLIENT_NAME_CONTENT_SHARING,
+        id: `${itemType}_${itemID}`,
+        token,
+    });
+
+    const getError = useCallback(
+        (error: $AxiosError<any> | ErrorResponseData) => {
+            let errorObject;
+            if (error.status) {
+                errorObject = contentSharingMessages[CONTENT_SHARING_ERRORS[error.status]];
+            } else if (error.response && error.response.status) {
+                errorObject = contentSharingMessages[CONTENT_SHARING_ERRORS[error.response.status]];
+            } else {
+                errorObject = contentSharingMessages.loadingError;
+            }
+
+            setErrorMessage(errorObject);
+        },
+        [setErrorMessage],
+    );
+
+    useEffect(() => {
+        setItem(null);
+        setSharedLink(null);
+        setCurrentUserID(null);
+    }, [token, itemID, itemType]);
+
+    useEffect(() => {
+        const getItemSuccess = (itemData: ContentSharingItemAPIResponse) => {
```

Comment:
Can this be moved up to a component-level variable? Should it be named something like `handleGetItemSuccess` to retain its status as a verb?

Reply:
I initially wrote this as a component-level function, but because it alters the component state, I had to wrap it in a `useCallback` hook. The Hooks FAQ recommends placing functions that reference props/state variables inside of effects instead of wrapping them in `useCallback`, so I moved it.